crates/common/tedge_config/src/tedge_config_repository.rs

```diff
@@ -32,7 +32,7 @@ impl ConfigRepository<TEdgeConfig> for TEdgeConfigRepository {
 
         // Create `$HOME/.tedge` or `/etc/tedge` directory in case it does not exist yet
         if !self.config_location.tedge_config_root_path.exists() {
-            let () = fs::create_dir(self.config_location.tedge_config_root_path())?;
+            let () = fs::create_dir_all(self.config_location.tedge_config_root_path())?;
```

Comment:
Did not get why do we need create_dir_all here. Because only one directory or a file has to be created nothing else inside right?

Reply:
So at least in [this test](https://github.com/thin-edge/thin-edge.io/blob/a4ffeccf60090e4456755bc53a6e3b8c8038e855/crates/common/tedge_config/src/tedge_config_location.rs#L59) there's two directories involved and if `/opt/etc/` does not exist, it fails.
It is just a change that is future-proof!

Reply:
If one uses them! :laughing: I ran into this while exploring how to run containerized tests where I did not have the opportunity run the post-install scripts...

Reply:
Hmm, right. that's why I said in the production case!.